clients/src/main/java/org/apache/kafka/common/Metric.java

```diff
@@ -27,8 +27,14 @@
     public MetricName metricName();
 
     /**
+     * The value of the metric as double if the metric is measurable
+     * @throws IllegalStateException if this metric does not have a measurable double value
+     */
+    public double measurableValue();
+
+    /**
      * The value of the metric
      */
-    public double value();
+    public Object value();
```

Comment:
Need to watch out for compatibility here, https://cwiki.apache.org/confluence/display/KAFKA/KIP-104%3A+Granular+Sensors+for+Streams made this public API

Reply:
@ewencp Thank you, I hadn't realized that `Metric` was a public API.